src/ngx_http_lua_req_body.c

```diff
@@ -303,8 +315,14 @@ ngx_http_lua_ngx_req_get_body_data(lua_State *L)
     buf = (u_char *) lua_newuserdata(L, len);
 
     p = buf;
-    for (cl = r->request_body->bufs; cl; cl = cl->next) {
-        p = ngx_copy(p, cl->buf->pos, cl->buf->last - cl->buf->pos);
+    rest = len;
+    for (cl = r->request_body->bufs; cl && rest; cl = cl->next) {
```

Comment:
for (cl = r->request_body->bufs; cl != NULL && rest > 0; cl = cl->next) {

Reply:
fixed